### PR TITLE
add 'parse' hook to add support for tweaking 'raw' easyconfig (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
see #2557 

cc @akesandgren

### CODE REVIEW

File: easybuild/framework/easyconfig/easyconfig.py [hunk 1]
```diff
@@ -446,8 +446,12 @@ def parse(self):
                 self.log.debug("Ignoring unknown easyconfig parameter %s (value: %s)" % (key, local_vars[key]))
 
         # trigger parse hook
+        # templating is disabled when parse_hook is called to allow for easy updating of mutable easyconfig parameters
+        # (see also comment in resolve_template)
         hooks = load_hooks(build_option('hooks'))
+        self.enable_templating = False
```
Comment (akesandgren) [inline easybuild/framework/easyconfig/easyconfig.py]: old_templating_state = self.enable_templating
and reset to old value after

**Comment by boegel** [comment]
@akesandgren Good to go?

**Comment by akesandgren** [comment]
Yeah i think so, can't find any problems with it.

**Comment by akesandgren** [comment]
@vanzod Any thoughts on this?

**Comment by boegel** [comment]
@ocaisa To answer the issue you raised during today's conf call: any changes made to the `EasyConfig` object in `parse_hook` are (currently) *not* included in the dumped easyconfig file in the installation directory or the easyconfigs archive (i.e. `--repositorypath`).

So, unless you make sure you have the hooks active every time, picking up a dumped easyconfig file later may result in surprises...

We can look into fixing that, but maybe some people wouldn't expect things to trickle down, I'm not quite sure (I can see problems with both approaches), and I'm not sure how easy it would be either to make things trickle down (the classic "there be dragons here" probably applies...).

**Comment by akesandgren** [comment]
I for one would expect it NOT to end up in the dumped one, but it might be good to have a comment added that hooks where changing stuff.

But if someone wants the changes to end up in the dump, this should probably be configurable.

**Comment by ocaisa** [comment]
@boegel I wouldn't have a problem with the archived easyconfig not including the changes but I think I would expect to see them in the `reprod` directory

**Comment by akesandgren** [comment]
Yes, that would be good, didn't think of that one.

**Comment by boegel** [comment]
follow-up on what @ocaisa mentioned in #2565

**Pull request merged**